CompanyPaymentsProject/Company.cs

```diff
@@ -6,41 +6,30 @@
 
 namespace CompanyPaymentsProject
 {
-    class Company : Person
+    class Company
     {
-        
-        //lista osób i możliwość ich dodawania
-        List<Person> people = new List<Person>()
+        private List<Person> _persons = new List<Person>();
+       
+        public List<Person> GetPersonList()
         {
-            new Person { PersonID = 1, Name = "Jan", Surname = "Kowalski", Post.Assistant},
-            new Person { PersonID = 2, Name = "Anna", Surname = "Nowak", Post.Junior},
-            new Person { PersonID = 3, Name = "John", Surname = "Doe", Post.Manager},
-        };
-        public void AddNewWorker()
+            return _persons;
+        }
+
+        public void AddPayCheck(int personId)
         {
-            for (int i = people.Count; i < people.Count+1; i++)
-            {
-                people[i] = new Person();
-                Console.WriteLine("Enter worker #{0} name", i + 1 + people.Count);
-                people[i].Name = Console.ReadLine();
-                Console.WriteLine("Enter worker #{0} Surname", i + 1 + people.Count);
-                people[i].Surname =Console.ReadLine();
-                Console.WriteLine("Set the worker post in the company: 1. Manager, 2. Junior, 3. Assistant");
-                int workerPost = int.Parse(Console.ReadLine());
+            var person = _persons.Where(x => x.PersonID == personId);            
+            var salary = new SalaryCalculator().CalculateSalary(person.Post);
+            person.Paychecks.Add(new Paycheck { Salary = salary });            
```

Comment:
Nie uzupełniasz daty i paycheckId - jeśli chodzi o drugie to ma to jakiś sens w tym konkretnym przypadku, bo nie zapisujemy do bazy danych i można się pogubić w nadawaniu indexów. Ale może warto jakąś datę sobie generować przypadkową (wpisz np. c# random date w googlach) i coś napisz takiego, ale w osobnej klasie np. RandomDateGenerator. Tutaj tylko to wywołasz i wpiszesz do zmiennej z datą.

Reply:
Albo zrób zupełnie inaczej: W tej funkcji dodaj parametr wejściowy DateTime date i w funkcji Main jak wywołasz AddPaycheck to tam sobie ustawisz datę, będzie w tym momencie więcej kontroli nad tym co się dzieje.